wallet-sdk/src/main/kotlin/org/stellar/walletsdk/exception/WalletException.kt

```diff
@@ -1,17 +1,31 @@
 package org.stellar.walletsdk.exception
 
 import okhttp3.Response
+import org.stellar.sdk.requests.ErrorResponse
+import org.stellar.walletsdk.util.toJson
+
+data class AnchorErrorResponse(val error: String)
 
 sealed class WalletException : Exception {
   constructor(message: String) : super(message)
   constructor(message: String, cause: Exception) : super(message, cause)
 }
 
-// TODO: add arguments for error code, response message, etc.
-class NetworkRequestFailedException(
-  rawResponse: Response,
-  label: String? = "Request failed",
-) : WalletException("$label: $rawResponse")
+open class ServerRequestFailedException(response: Response) :
+  WalletException("Anchor request failed") {
+  private val errorResponse: AnchorErrorResponse = response.toJson()
```

Comment:
parsing response can potentially throw, can you wrap it in try/catch and return null on error?
(better: add toJsonOrNull function)